Zend/zend_ast.c

```diff
@@ -1616,15 +1616,26 @@ static ZEND_COLD void zend_ast_export_ex(smart_str *str, zend_ast *ast, int prio
 		}
 
 		case ZEND_AST_CONST_DECL:
-		case ZEND_AST_CLASS_CONST_DECL:
 			smart_str_appends(str, "const ");
 			goto simple_list;
 		case ZEND_AST_CLASS_CONST_GROUP:
 			if (ast->child[1]) {
 				zend_ast_export_attributes(str, ast->child[1], indent, 1);
 			}
-			zend_ast_export_ex(str, ast->child[0], 0, indent);
-			break;
+
+			if (ast->attr & ZEND_ACC_PUBLIC) {
+				smart_str_appends(str, "public ");
+			} else if (ast->attr & ZEND_ACC_PROTECTED) {
+				smart_str_appends(str, "protected ");
+			} else if (ast->attr & ZEND_ACC_PRIVATE) {
+				smart_str_appends(str, "private ");
+			}
```

Comment:
This code is now repeated three times. Can you please extract it into a function? zend_ast_export_visibility or so.